module/module.go

```diff
@@ -617,7 +617,7 @@ func (m *Module) ReconfigureResource(ctx context.Context, req *pb.ReconfigureRes
 		return nil, err
 	}
 
-	m.logger.Debugw("rebuilding", "name", conf.ResourceName())
+	m.logger.Debugw("rebuilding", "name", conf.ResourceName().String())
```

Comment:
[drive-by] Structure-logging this resource name object resulted in

```
server.go:453: 2025-06-17T12:37:00.294-0400	DEBUG	optional-deps	module/module.go:620	rebuilding	{"nameError":"PANIC=interface conversion: map[string]interface {} is not fmt.Stringer: missing method String","log_ts":"2025-06-17T16:37:00.294Z"}
```
Just stringified for now. Was this an issue we were aware of, @dgottlieb ? I forget.

Reply:
This is an issue we've seen. I thought this was fixed by either something Cheuk or Josh did. But I think this is due to the `map[string]interface` getting zap encoded over our log entry proto as a "string" type.

Such that the zap encoding code assumes its a `fmt.Stringer` [here](https://github.com/uber-go/zap/blob/07077a697f639389cc998ff91b8885feb25f520d/zapcore/field.go#L231).